services/blockchain-indexer/shared/database/schema/eventTopics.js

```diff
@@ -15,8 +15,9 @@
  */
 module.exports = {
 	tableName: 'event_topics',
-	primaryKey: ['id', 'topic'],
+	primaryKey: 'tempID',
 	schema: {
+		tempID: { type: 'string' },
```

Comment:
Would naming it `id_topic` be more expressive? Also, why do we need to store it separately since both `id` and `topic` are stored already?

Reply:
This is necessary because of the implementation of our current MySQL abstraction. The `count` method fails if the primary key is composite. This should be fixed with the issue: https://github.com/LiskHQ/lisk-service/issues/1188

Reply:
Is it feasible to write raw queries instead of storing redundant data?

Reply:
I'd like to minimize the changes/rework that would be necessary everywhere in the code once https://github.com/LiskHQ/lisk-service/issues/1188 is addressed. Hence, avoiding the raw queries until and unless that is the only way to go ahead.

Reply:
I get your point. My only concern is changing DB structure will probably require downtime and will require unnecessary resources until that card is merged. Whereas code changes won't have an additional impact on production downtime.

Reply:
Thank you for the concern but I'll take care of coordinating and handling the situation when the time comes 🙂